src/julia_threads.h

```diff
@@ -214,6 +215,68 @@ typedef struct _jl_tls_states_t {
 #endif
 } jl_tls_states_t;
 
+#define JL_RNG_SIZE 5 // xoshiro 4 + splitmix 1
+
+// all values are callable as Functions
+typedef jl_value_t jl_function_t;
+typedef struct _jl_timing_block_t jl_timing_block_t;
+typedef struct _jl_timing_event_t jl_timing_event_t;
+typedef struct _jl_excstack_t jl_excstack_t;
+typedef struct _jl_handler_t jl_handler_t;
+
+typedef struct _jl_task_t {
```

Comment:
A major chunk of this PR seems to be about moving `_jl_task_t` from `julia.h` to here. Is there a functional reason why this is necessary for this PR? Perhaps something got added to `_jl_task_t` -- but it's difficult to say because everything is in a single commit

Reply:
Made PR #57117 to take care of this